packages/core/src/consumed-thing.ts

```diff
@@ -491,7 +513,8 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
                     "[core/consumed-thing]",
                     `ConsumedThing '${this.title}' chose cached client for '${schemes[cacheIdx]}'`
                 );
-                client = this.getClients().get(schemes[cacheIdx]);
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- if cacheIdx is valid, then clients *contains* schemes[cacheIdx]
+                client = this.getClients().get(schemes[cacheIdx])!;
```

Comment:
Would it make sense to throw an error here instead?

Reply:
The thing is that at the point we are sure that this.getClients().get(schemes[cacheIdx])! will not be undefined. Check line 508 cacheIdx will be not be -1 if this.getClients() contains that scheme.